### PR TITLE
Feat/remove headers from uploaded wordlists via the Google Sheets API

### LINKED ISSUE #223: Remove headers from uploaded wordlists via the Google Sheets API
An extension of issue #199
> Remove headers such as "word" and "count" headers in the first row of all uploaded wordlists (either Google Sheets or Excel).
> 
> This should be done automatically.

Add functionality from #217 to the Google Sheets API. Code reuse is ideal.

### PR DESCRIPTION
**Pull request by yehee**
Resolves #223

### CODE REVIEW

File: src/app/components/GoogleSheetsInput.svelte
Comment (eddieantonio) [inline src/app/components/GoogleSheetsInput.svelte]: Interesting! So the old regex was too restrictive? How did you come up with this regex?

Also, `{1}` is always redundant — it means select the previous token exactly once, which is the default:

```suggestion
      "http(s?)://docs.google.com/spreadsheets/[a-z]/[\\w-]+/[\\w#=-]+"
```

**Review by eddieantonio — APPROVED** [review]
LGTM! Just one minor suggestion, so I will let you hit the merge button when ready.

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [inline src/app/components/GoogleSheetsInput.svelte:10]: What are your thoughts on keeping this line in more or less?
I like how the naming is now `CLIENT_ID` rather than `G_CLIENT_ID`, but I liked how this line explicitly states that `CLIENT_ID` comes from an env file.

I'm neutral on this, so feel free to do what you think looks best

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [inline src/app/components/GoogleSheetsInput.svelte:10]: Do you think that we should make a new file for constants like these, now that it's used in multiple places?

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [inline src/app/components/GoogleSheetsInput.svelte:131]: The behaviour looks like it matches the Excel file handling, thanks!

I think we should make a new ticket to try to run all the file types into the same business rules so that we don't need to make 3+ changes for each new feature. What are your thoughts?

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [inline src/app/components/GoogleSheetsInput.svelte]: Out of curiosity, what's does this different pattern do compared to the old pattern?

**Review by ChrisChrisLoLo — APPROVED** [review]
Looks good to me!

Great eye for detail, the code here looks very similar to the Excel reader.

I'll let you decide if you want to merge after reading my comments and suggestions. If you want a re-review feel free to let me know.

I would have requested that you added a test for this, but currently, we have not yet set that up (#198), and it could take a fair amount of time considering that we need an API key for automation. I want to avoid Unhelpful behaviour # 3 in this article: https://medium.com/@sandya.sankarram/unlearning-toxic-behaviors-in-a-code-review-culture-b7c295452a3c. We should probably get the testing going for it soon though.

File: src/app/components/GoogleSheetsInput.svelte
Comment (yehee) [reply src/app/components/GoogleSheetsInput.svelte]: I noticed that mine includes underscores and dashes, so I've included them: `https://docs.google.com/spreadsheets/d/11NohN7t_iZ1hRx-0J7Qtr7ZMVlGl0ffjui5HLqzwhmU`. I'd love to take a look if there's documentation that I can refer back to!

File: src/app/components/GoogleSheetsInput.svelte
Comment (yehee) [reply src/app/components/GoogleSheetsInput.svelte:10]: Ahh I can definitely see that! However with your documentation on client/API keys, I'm hoping that should be clear in the devs perspective, what do you think?

File: src/app/components/GoogleSheetsInput.svelte
Comment (yehee) [reply src/app/components/GoogleSheetsInput.svelte:10]: Yes, but I also noticed that yours wasn't zero-based i.e. WORD = 1. So they unfortunately do not share the same value..

File: src/app/components/GoogleSheetsInput.svelte
Comment (yehee) [reply src/app/components/GoogleSheetsInput.svelte:131]: It would be really nice, might need to do some investigation before since each of their implementations differs by a lot!

File: src/app/components/GoogleSheetsInput.svelte
Comment (yehee) [reply src/app/components/GoogleSheetsInput.svelte]: This part `[a-zA-z0-9]+` from the old one was a little more strict in that it didn't allow underscores or dashes. Other changes are just to make it a bit simpler!

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [reply src/app/components/GoogleSheetsInput.svelte:131]: Good point, I'll make some investigation ticket to start a discussion. There might be an OOP pattern that could help us

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [reply src/app/components/GoogleSheetsInput.svelte:10]: Good catch

File: src/app/components/GoogleSheetsInput.svelte
Comment (ChrisChrisLoLo) [reply src/app/components/GoogleSheetsInput.svelte:10]: Yeah, I definitely see what you're saying. Both styles can be figured out by the developer using `ctrl`+`shift`+`F`. Considered me swayed!

File: src/app/components/GoogleSheetsInput.svelte
Comment (eddieantonio) [reply src/app/components/GoogleSheetsInput.svelte:131]: [Template Method Pattern!](https://en.wikipedia.org/wiki/Template_method_pattern)

**Pull request merged**
